lib/models/user-identity.js

```diff
@@ -112,14 +114,23 @@
       var userModel = (userIdentityModel.relations.user &&
                        userIdentityModel.relations.user.modelTo) ||
                        loopback.getModelByType(loopback.User);
-      var userObj = (options.profileToUser || profileToUser)(provider, profile);
-      if (!userObj.email) {
-        return cb('email is missing from the user profile');
+      var userObj = (options.profileToUser || profileToUser)(provider, profile, options);
+      if (!userObj.email && !options.emailOptional) {
+        process.nextTick(function() {
```

Comment:
@alexpit You are correct, this is indeed a bug. There is a missing return before the process.nextTick.

Can you file an issue and maybe create a PR?